File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    int combination(int n, int r)
    {
        int result=1;

        for(int i=0;i<r;i++)
        {
            result=result*(n-i);
            result=result/(i+1);
        }
        return result;
    }
    vector<vector<int>> generate(int numRows) {

        vector<vector<int>> ans;

        for(int r=1;r<=numRows;r++)
        {
            vector <int> p;

            for(int c=1;c<=r;c++)
            {
                p.push_back(combination(r-1,c-1));
            }
            ans.push_back(p);
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {

    int find_fact(int n, int r)
    {
        int result=1;
        for(int i=0;i<r;i++)
        {
            result=result*(n-i);
            result=result/(i+1);
        }

        return result;
    }
    public List<List<Integer>> generate(int numRows) {

        List<List<Integer>> ans = new ArrayList<>();

        for(int i=1;i<=numRows;i++)
        {
            List<Integer> sub= new ArrayList<>();
            for(int j=1;j<=i;j++)
            {
                sub.add(find_fact(i-1,j-1));
            }
            ans.add(sub);
        }

        return ans;
    }
}